exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestNestedLoopJoin.java

```diff
@@ -333,4 +339,55 @@ public void testNlJoinWithLargeRightInputSuccess() throws Exception {
       test(RESET_JOIN_OPTIMIZATION);
     }
   }
+
+  private static void buildFile(String fileName, String[] data, File testDir) throws IOException {
+    try(PrintWriter out = new PrintWriter(new FileWriter(new File(testDir, fileName)))) {
+      for (String line : data) {
+        out.println(line);
+      }
+    }
+  }
+
+  public static void testWithEmptyJoin(File testDir, String joinType,
+                                String joinPattern, long result) throws Exception {
+    buildFile("dept.json", new String[0], testDir);
+    String query = String.format(testEmptyJoin, joinType);
+    testPlanMatchingPatterns(query, new String[]{joinPattern}, new String[]{});
+    testBuilder()
+            .sqlQuery(query)
+            .unOrdered()
+            .baselineColumns("cnt")
+            .baselineValues(result)
+            .build().run();
+  }
+
+  @Test
+  public void testNestedLeftJoinWithEmptyTable() throws Exception {
+    try {
+      test(DISABLE_NLJ_SCALAR);
```

Comment:
Please use `client.setSessionOption(...)` to make it clear what we are doing here.